node/network/statement-distribution/src/lib.rs

```diff
@@ -502,6 +506,19 @@ impl PeerData {
 			.check_can_receive(fingerprint, max_message_count)
 	}
 
+	/// Receive a notice about out of view statement and returns the value of the old flag
+	fn receive_out_of_view_statement(&mut self, relay_parent: &Hash) -> bool {
+		self.view_knowledge
+			.get_mut(relay_parent)
+			.map_or(false, |relay_parent_peer_knowledge| {
+				let old = relay_parent_peer_knowledge.seen_out_of_view;
+				if !old {
+					relay_parent_peer_knowledge.seen_out_of_view = true;
+				}
+				old
```

Comment:
```suggestion
				let old = relay_parent_peer_knowledge.seen_out_of_view;
				relay_parent_peer_knowledge.seen_out_of_view = true;
				old
```

Reply:
Well, branch predictor success (which will be successfull in all branches but the first one) is almost free on x86, whilst unconditionally set `relay_parent_peer_knowledge.seen_out_of_view = true;` emits a store instruction all the time that is not free :) But I agree that it's a completely negligible micro-optimization.

Reply:
Code simplification > micro-opt.

I think we don't really require this optimization to only log it once, as long as it's trace.